composer/trainer/trainer.py

```diff
@@ -3327,12 +3337,14 @@ def _eval_loop(
                                                 outputs.append(v)
                                     else:
                                         outputs = self.state.outputs.cpu()
+                                    batch = DeviceCPU().batch_to_device(batch)
```

Comment:
```suggestion
                                    batch = DeviceCPU().batch_to_device(self.state.batch)
```

Reply:
Thank you this is now integrated in this PR